### PR TITLE
Update Microsoft.VSSDK.BuildTools from 17.0.1600 to 17.12.2069

### LINKED ISSUE #13832: Unable to build NuGet.Client VSIX - "GetDeploymentPathFromVsixManifest" task failed unexpectedly
I'm not able to build the NuGet.Client VSIX with the latest 17.12 Preview 3 Visual Studio Internal Preview. 

> The "GetDeploymentPathFromVsixManifest" task failed unexpectedly.
> System.AggregateException->Microsoft.Build.Framework.BuildException.GenericBuildTransferredException: One or more errors occurred. ---> Microsoft.Build.Framework.BuildException.GenericBuildTransferredException:
> Object reference not set to an instance of an object.
>    --- End of inner exception stack trace ---

What I've tracked down is that we target a very old 17.0.x version of Microsoft.VSSDK.BuildTools https://github.com/NuGet/NuGet.Client/blob/db35f736d692fe4a46357a3a2de1ea6206dddeb3/Directory.Packages.props#L79

Likely there's some breaking change that's occurred between 17.0 and 17.12.

### PR DESCRIPTION
**Pull request by donnie-msft**
# Bug

Fixes:  https://github.com/NuGet/Home/issues/13832

## Description

It seems we don't regularly update Microsoft.VSSDK.BuildTools and therefore we have a very old 17.0.x version of that dependency https://github.com/NuGet/NuGet.Client/blob/db35f736d692fe4a46357a3a2de1ea6206dddeb3/Directory.Packages.props#L79

- Update `Microsoft.VSSDK.BuildTools` from 17.0.1600 to 17.12.2069
  - Note that I did try some slightly older packages from 17.11.x, but experienced vulnerable transitive dependencies and decided the latest stable version was best to work towards.
- Add reference to the new `Microsoft.VisualStudio.ExtensionEngineContract`
- Add/Remove assemblies from the VSIX as necessary from the change to dependent assemblies

Likely, there was some breaking change that's occurred between 17.0 and 17.12.

### Side effects

- The VSSDK was now finding `Lucene.net` from a different directory via a Project Reference which was not signed, and therefore failed our signing verification. @jeffkl added a step to remove the unsigned version of this assembly when it's being pulled from a Project Reference.
- The VSSDK was now adding all localization resource assemblies which it did not do previously. This caused the wrong root path to be used for localized assemblies. @jeffkl put in a removal of all of those `*resources.dll` assemblies items just before our build adds them as it has historically with the `GetLocalizedDlls` Target that follows.

#### VSIX content changes
- Our VSIX no longer includes a `NuGet.VisualStudio.Client.clientenabledpkg` since the CodeSpaces functionality for this was moved to GitHub Codespaces by 2022
- The VSSDK is now including generated XML Documentation files for assemblies. I've added this to the ignore list since we do not want to ship them. Our build logic now considers .XML file extensions when loading files for the vsixinclude/vsixignore comparison.
- Include in the VSIX the `NuGet.PackageManagement.PowerShellCmdlets.dll-Help.xml` which is the only XML file we ship. It appears to be used for PowerShell Console help (eg, get-help install-package).

##### Delta in our VSIX content with this PR:

|Removed from VSIX|
|--|
|NuGet.VisualStudio.Client.clientenabledpkg|
|System.Buffers.dll|
|System.Memory.dll|
|System.Numerics.Vectors.dll|
|System.Runtime.CompilerServices.Unsafe.dll|

Flattened directory content files used for comparison:
[dev branch files.txt](https://github.com/user-attachments/files/17403515/NuGet.Toolsdev20241015.txt)
[this branch fe0f5e1ed files.txt](https://github.com/user-attachments/files/17403520/NuGet.Toolsdev20241016.txt)

Special thanks to @jeffkl for helping address the above side effects with his MS Build expertise :)

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [x] Added tests - N/A - I tested that the VSIX builds, installs, and can open PM UI.
- [x] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

**Review by jeffkl — DISMISSED** [review]
🎉

**Review by jebriede — DISMISSED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (jeffkl) [inline src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj]: Let's add Lucene.Net to the ignore list with a comment saying "ignore the Lucene.Net that the VS SDK adds because this project will manually add the signed one"

Also, remove the Remove below (lines 344-355).  Looks like this whole file can be undone technically.

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj]: Ok, I've ignored Lucene.net, but I'm not following how we can remove the change we made together offline to remove the VSIXItems for already added .resources.dlls. How would that work now?

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj]: Talked offline and we can ignore Lucene.net from this calculation since elsewhere in the build we add it from `GetThirdPartyAssemblies`.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: I'd just like to confirm if someone downloaded the vsix from the dev branch, and from this branch, and validated that the files in both vsix's are the same?

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: I did, and they are not. See these commits:
image

Comment (zivkan) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: Is the PR ready to be reviewed and merged, or is it still being worked on while the file differences are still being worked on? 

When this PR pops up in my notification list, I'm not sure if I should review and approve, or if I should wait.

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: It is ready for review. Sorry for the noise in the past day or so, we kept thinking it was done but little things kept cropping up. It appears my build is green which is why I'm updating the description now.

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: I added a nifty table.

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: > When this PR pops up in my notification list, I'm not sure if I should review and approve, or if I should wait.

Open to suggestions. Even if I had put it back to a draft, you'll still get those notifications as GitHub does not unsubscribe anyone.

**Review by jeffkl — APPROVED** [review]
🎉

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (zivkan) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:275]: yes, but if I see a notification for a draft PR, I can immediately move on, knowing I'll get another notification when it's ready for review, at which time I should look at the diff.

**Review by zivkan — APPROVED** [review]

**Pull request merged**
